### PR TITLE
Covid testing

### PR DESCRIPTION
**Pull request by IsaiahV98**

### CODE REVIEW

File: libs/covid/data-api/src/lib/modules/counties/counties.controller.spec.ts [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        CountiesService, 
+        { provide: getRepositoryToken(County), useClass: Repository }, 
+        { provide: getRepositoryToken(CountyClaim), useClass: Repository },
+        { provide: getRepositoryToken(StatusType), useClass: Repository },
+        { provide: getRepositoryToken(EntityStatus), useClass: Repository }, 
+        { provide: getRepositoryToken(TestingSite), useClass: Repository }
```
Comment (Proantagonist) [inline libs/covid/data-api/src/lib/modules/counties/counties.controller.spec.ts:19]: These are unecessary

File: libs/covid/data-api/src/lib/modules/counties/counties.controller.spec.ts [hunk 2]
```diff
@@ -1,11 +1,23 @@
 import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { County, CountyClaim, StatusType, EntityStatus, TestingSite } from '@tamu-gisc/covid/common/entities';
```
Comment (Proantagonist) [inline libs/covid/data-api/src/lib/modules/counties/counties.controller.spec.ts:4]: Clear unused imports: StatusType, EntityStatus, TestingSite

File: libs/geoservices/covid/src/lib/pages/resources/pages/testing-sites/testing-sites.component.spec.ts [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ TestingSitesComponent ]
-    })
-    .compileComponents();
+      imports: [HttpClientTestingModule, FormsModule, ReactiveFormsModule, UIFormsModule, RouterTestingModule, StorageServiceModule],
+      declarations: [ TestingSitesComponent ],
+      providers: [
+        TestingSitesComponent,{
```
Comment (Proantagonist) [inline libs/geoservices/covid/src/lib/pages/resources/pages/testing-sites/testing-sites.component.spec.ts:21]: Component as provider

File: libs/geoservices/covid/src/lib/pages/resources/resources.component.spec.ts [hunk 4]
```diff
@@ -1,13 +1,16 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { ResourcesComponent } from './resources.component';
+import { RouterTestingModule } from '@angular/router/testing';
+
```
Comment (Proantagonist) [inline libs/geoservices/covid/src/lib/pages/resources/resources.component.spec.ts:5]: Prettier and  import ordering

File: libs/geoservices/data-access/src/lib/guards/auth.guard.spec.ts [hunk 5]
```diff
@@ -1,15 +1,29 @@
...
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { EnvironmentModule, env } from '@tamu-gisc/common/ngx/environment';
+import { SettingsModule } from '@tamu-gisc/common/ngx/settings';
 
 describe('AuthGuard', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      providers: [AuthGuard]
+      imports: [HttpClientTestingModule, RouterTestingModule, EnvironmentModule, SettingsModule],
```
Comment (Proantagonist) [inline libs/geoservices/data-access/src/lib/guards/auth.guard.spec.ts:12]: SettingsModule is not necessary

File: libs/geoservices/data-access/src/lib/guards/auth.guard.spec.ts [hunk 6]
```diff
@@ -1,15 +1,29 @@
...
+      
     });
   });
 
   it('should ...', inject([AuthGuard], (guard: AuthGuard) => {
     expect(guard).toBeTruthy();
   }));
+
 });
+
```
Comment (Proantagonist) [inline libs/geoservices/data-access/src/lib/guards/auth.guard.spec.ts:29]: Prettier

File: .vscode/settings.json [hunk 7]
```diff
@@ -33,5 +33,6 @@
     "yml"
   ],
   "jest.autoEnable": false,
-  "cSpell.enabled": true
-}
\ No newline at end of file
+  "cSpell.enabled": true,
+  "editor.formatOnSave": true
```
Comment (Proantagonist) [inline .vscode/settings.json:37]: Keep these settings in your user VSCode user configuration.

File: libs/covid/data-api/src/lib/modules/websites/websites.controller.spec.ts [hunk 8]
```diff
@@ -1,18 +1,32 @@
...
+      ],
+      controllers: [WebsitesController]
     }).compile();
 
     controller = module.get<WebsitesController>(WebsitesController);
   });
 
   it('should be defined', () => {
-    expect(controller).toBeDefined();
+    expect(controller).toBeDefined;
```
Comment (Proantagonist) [inline libs/covid/data-api/src/lib/modules/websites/websites.controller.spec.ts:30]: This still needs to be a method call, not just a reference.

In other words, missing `()`, like `toBeDefined()`

File: libs/covid/data-api/src/lib/modules/testing-sites/testing-sites.service.spec.ts [hunk 9]
```diff
@@ -1,12 +1,41 @@
...
+        { provide: getRepositoryToken(County), useClass: Repository },
+        { provide: getRepositoryToken(CountyClaim), useClass: Repository },
+        { provide: getRepositoryToken(TestingSite), useClass: Repository },
+        { provide: getRepositoryToken(TestingSiteInfo), useClass: Repository },
+        { provide: getRepositoryToken(Location), useClass: Repository },
+        { provide: getRepositoryToken(User), useClass: Repository },
+        { provide: getRepositoryToken(EntityStatus), useClass: Repository },
+        { provide: getRepositoryToken(CountyClaimInfo), useClass: Repository },
+        { provide: getRepositoryToken(EntityToValue), useClass: Repository },
+        CountyClaimsService
```
Comment (Proantagonist) [inline libs/covid/data-api/src/lib/modules/testing-sites/testing-sites.service.spec.ts:37]: Group references and objects together.

File: libs/cpa/common/ngx/src/lib/modules/create/components/scenarios-list/scenarios-list.component.spec.ts [hunk 10]
```diff
@@ -1,16 +1,42 @@
...
-      declarations: [ ScenariosListComponent ]
-    })
-    .compileComponents();
+      imports: [
+        CommonModule,
+        RouterTestingModule,
+        UILayoutModule,
+        NavigationBreadcrumbModule,
+        HttpClientTestingModule,
+        EnvironmentModule
```
Comment (Proantagonist) [inline libs/cpa/common/ngx/src/lib/modules/create/components/scenarios-list/scenarios-list.component.spec.ts:26]: Unecessary imports:

- CommonModule,
- UILayoutModule

File: libs/cpa/common/ngx/src/lib/modules/create/components/scenarios-list/scenarios-list.component.spec.ts [hunk 11]
```diff
@@ -1,16 +1,42 @@
...
+      declarations: [ScenariosListComponent],
+      providers: [
+        {
+          provide: env,
+          useValue: { api_url: 'https://' }
+        },
+        {
+          provide: ActivatedRoute,
+          useValue: { params: of({ route: '' }) }
+        }
```
Comment (Proantagonist) [inline libs/cpa/common/ngx/src/lib/modules/create/components/scenarios-list/scenarios-list.component.spec.ts:37]: Unnecessary provider. Only needed when `ActivateRoute` is injected into the component.

File: libs/cpa/common/ngx/src/lib/modules/forms/components/workshop-builder/workshop-builder.component.spec.ts [hunk 12]
```diff
@@ -8,9 +15,70 @@ describe('WorkshopBuilderComponent', () => {
...
+        ScenariosModule
+      ],
+      declarations: [WorkshopBuilderComponent],
+      providers: [
+        {
+          provide: env,
+          useValue: { api_url: 'workshops' }
+        }
+      ]
+    }).compileComponents();
```
Comment (Proantagonist) [inline libs/cpa/common/ngx/src/lib/modules/forms/components/workshop-builder/workshop-builder.component.spec.ts:81]: Meant to remove this?

File: libs/geoservices/api/src/lib/components/response-viewer/response-viewer.component.spec.ts [hunk 13]
```diff
@@ -1,25 +1,18 @@
...
 describe('ResponseViewerComponent', () => {
-  let component: ResponseViewerComponent;
-  let fixture: ComponentFixture<ResponseViewerComponent>;
-
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ ResponseViewerComponent ]
-    })
-    .compileComponents();
+      imports: [HighlightPlusModule, CommonModule],
```
Comment (Proantagonist) [inline libs/geoservices/api/src/lib/components/response-viewer/response-viewer.component.spec.ts:10]: `CommonModule` unnecessary. 

Clean up orphan imports.

File: libs/geoservices/api/src/lib/components/response-viewer/response-viewer.component.spec.ts [hunk 14]
```diff
@@ -1,25 +1,18 @@
...
   }));
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(ResponseViewerComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
-
-  it('should create', () => {
+  it('should create', inject([ResponseViewerComponent], (component: ResponseViewerComponent<{}, {}>) => {
```
Comment (Proantagonist) [inline libs/geoservices/api/src/lib/components/response-viewer/response-viewer.component.spec.ts:15]: Why inject vs getting component instance from fixture?

```js
  beforeEach(() => {
    fixture = TestBed.createComponent(ComponentToken);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });
```

I'm not saying this implementation is wrong, I just want to understand why.

File: libs/geoservices/api/src/lib/geoservices-api.component.spec.ts [hunk 15]
```diff
@@ -8,6 +14,7 @@ describe('GeoservicesApiComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
+      imports: [CommonModule, UILayoutModule, UIScrollToModule, RouterTestingModule, BrowserAnimationsModule],
```
Comment (Proantagonist) [inline libs/geoservices/api/src/lib/geoservices-api.component.spec.ts:17]: From what I've been able to dig up, `CommonModule` isn't required for tests? Tests pass without it.

Clean orphan import statement.

File: libs/geoservices/core/ngx/src/lib/components/listable/submissions/county/county.component.spec.ts [hunk 16]
```diff
@@ -8,9 +11,16 @@ describe('CountyComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ CountyListComponent ]
-    })
-    .compileComponents();
+      imports: [EnvironmentModule, HttpClientTestingModule],
+      declarations: [CountyListComponent],
+      providers: [
+        CountyListComponent,
```
Comment (Proantagonist) [inline libs/geoservices/core/ngx/src/lib/components/listable/submissions/county/county.component.spec.ts:17]: Components are *almost always* never providers. I can't think of a case where they would be, but there might be a reason to. To date, this should not be the case for any of our components.

File: libs/geoservices/covid/src/lib/pages/resources/pages/testing-sites/pages/create/create.component.spec.ts [hunk 17]
```diff
@@ -1,18 +1,40 @@
...
 import { CreateComponent } from './create.component';
+import { ReactiveFormsModule } from '@angular/forms';
+import { UIFormsModule } from '@tamu-gisc/ui-kits/ngx/forms';
+import { RouterTestingModule } from '@angular/router/testing';
+import { GeoservicesCoreNgxModule } from '@tamu-gisc/geoservices/core/ngx';
+import { EnvironmentModule, env } from '@tamu-gisc/common/ngx/environment';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { AppStorage } from '@tamu-gisc/common/ngx/local-store';
+import { StorageServiceModule, LOCAL_STORAGE } from 'angular-webstorage-service';
+
```
Comment (Proantagonist) [inline libs/geoservices/covid/src/lib/pages/resources/pages/testing-sites/pages/create/create.component.spec.ts:12]: Import ordering

File: libs/geoservices/internal/src/lib/geoservices-internal.component.spec.ts [hunk 18]
```diff
@@ -8,9 +13,9 @@ describe('InternalComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ InternalComponent ]
-    })
-    .compileComponents();
+      imports: [RouterTestingModule, UIFormsModule, UILayoutModule, BrowserAnimationsModule],
```
Comment (Proantagonist) [inline libs/geoservices/internal/src/lib/geoservices-internal.component.spec.ts:16]: `UIFormsModule` not necessary. 

Clean up orphan import statements.

File: libs/geoservices/internal/src/lib/pages/account/details/details.component.spec.ts [hunk 19]
```diff
@@ -1,35 +1,32 @@
...
+        }
+      ],
+      declarations: [DetailsComponent]
+    }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(DetailsComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
```
Comment (Proantagonist) [inline libs/geoservices/internal/src/lib/pages/account/details/details.component.spec.ts:19]: `detectChanges()` triggers Angular's change detection, including life cycle hooks. Without it, members like `ngOnInit` will not execute.

Avoid removing this statement.

File: libs/geoservices/internal/src/lib/pages/account/security/security.component.spec.ts [hunk 20]
```diff
@@ -1,32 +1,33 @@
...
+        }
+      ],
+      declarations: [SecurityComponent]
+    }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(SecurityComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
```
Comment (Proantagonist) [inline libs/geoservices/internal/src/lib/pages/account/security/security.component.spec.ts:19]: Re-introduce `detectChanges()`

File: libs/geoservices/public/src/lib/pages/services/services.component.spec.ts [hunk 21]
```diff
@@ -8,9 +12,9 @@ describe('ServicesComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ ServicesComponent ]
-    })
-    .compileComponents();
+      imports: [CommonModule, RouterTestingModule, UILayoutModule],
```
Comment (Proantagonist) [inline libs/geoservices/public/src/lib/pages/services/services.component.spec.ts:15]: CommonModule

File: libs/geoservices/covid/src/lib/pages/resources/pages/testing-sites/pages/create/create.component.spec.ts [hunk 22]
```diff
@@ -1,40 +1,47 @@
...
+        UIFormsModule,
+        RouterTestingModule,
+        StorageServiceModule,
+        GeoservicesCoreNgxModule,
+        EnvironmentModule,
+        HttpClientTestingModule
+      ],
+      declarations: [CreateComponent],
       providers: [
         CreateComponent,
```
Comment (Proantagonist) [inline libs/geoservices/covid/src/lib/pages/resources/pages/testing-sites/pages/create/create.component.spec.ts:32]: Components are almost never providers

File: libs/maps/mapbox/src/lib/maps-mapbox.module.spec.ts [hunk 23]
```diff
@@ -1,14 +1,6 @@
 import { async, TestBed } from '@angular/core/testing';
 import { MapsMapboxModule } from './maps-mapbox.module';
```
Comment (Proantagonist) [inline libs/maps/mapbox/src/lib/maps-mapbox.module.spec.ts]: Whitespace between import types

File: libs/ui-kits/ngx/forms/src/lib/components/textbox/textbox.component.spec.ts [hunk 24]
```diff
@@ -11,7 +11,7 @@ describe('TextboxComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [CommonModule, FormsModule, RouterTestingModule],
+      imports: [FormsModule],
```
Comment (Proantagonist) [inline libs/ui-kits/ngx/forms/src/lib/components/textbox/textbox.component.spec.ts:14]: Cleanup unused imports

File: libs/cpa/common/ngx/src/lib/modules/create/components/scenarios-list/scenarios-list.component.spec.ts [hunk 25]
```diff
@@ -1,10 +1,8 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { RouterTestingModule } from '@angular/router/testing';
-import { CommonModule } from '@angular/common';
 import { ActivatedRoute } from '@angular/router';
```
[the remark this answers is not in the document]
Comment (IsaiahV98) [reply libs/cpa/common/ngx/src/lib/modules/create/components/scenarios-list/scenarios-list.component.spec.ts]: Fixed

File: libs/geoservices/covid/src/lib/pages/resources/pages/testing-sites/pages/create/create.component.spec.ts (hunk 22 above)
Comment (IsaiahV98) [reply libs/geoservices/covid/src/lib/pages/resources/pages/testing-sites/pages/create/create.component.spec.ts:34]: Mistake on my end, meant to remove.

File: libs/maps/mapbox/src/lib/maps-mapbox.module.spec.ts (hunk 23 above)
Comment (IsaiahV98) [reply libs/maps/mapbox/src/lib/maps-mapbox.module.spec.ts]: Done

File: libs/ui-kits/ngx/forms/src/lib/components/textbox/textbox.component.spec.ts (hunk 24 above)
Comment (IsaiahV98) [reply libs/ui-kits/ngx/forms/src/lib/components/textbox/textbox.component.spec.ts:14]: Done.

File: libs/maps/mapbox/src/lib/components/mapbox-map/mapbox-map.component.spec.ts [hunk 26]
```diff
@@ -2,15 +2,28 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
...
+jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
+  GeolocateControl: jest.fn(),
+  Map: jest.fn(() => ({
+    addControl: jest.fn(),
+    on: jest.fn(),
+    remove: jest.fn()
+  })),
+  NavigationControl: jest.fn()
+}));
+
```
[the remark this answers is not in the document]
Comment (IsaiahV98) [reply libs/maps/mapbox/src/lib/components/mapbox-map/mapbox-map.component.spec.ts]: Moved mock to setupJestMock.ts under the test file.

File: libs/maps/mapbox/src/lib/maps-mapbox.module.spec.ts [hunk 27]
```diff
@@ -1,6 +1,14 @@
...
 
+/* 
+Mocked due to issue related to window.URL.createObjectURL https://github.com/mapbox/mapbox-gl-js/issues/3436
+*/
+
+jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
+  Map: () => ({})
+}));
+
```
[the remark this answers is not in the document]
Comment (IsaiahV98) [reply libs/maps/mapbox/src/lib/maps-mapbox.module.spec.ts]: Moved mock to setupJestMock.ts under the test file.

**Pull request merged**